### PR TITLE
Remove exists check when creating a file in S3

### LINKED ISSUE #103: PrestoS3FileSystem breaks read-after-write strict S3 consistency
Method: `com.facebook.presto.hive.s3.PrestoS3FileSystem#create`
will check for file existence when `overwrite==false`. File existence check will break read-after-write strict S3 consistency per S3 doc:
```
Amazon S3 provides read-after-write consistency for PUTS of new objects in
your S3 bucket in all regions with one caveat. The caveat is that if you make a
HEAD or GET request to the key name (to find if the object exists) before
creating the object, Amazon S3 provides eventual consistency for read-after-write. 
```

The existence check itself is weak as the file might be created while data is gathered in staging directory for upload.

I propose to remove the check of use `LIST` S3 request instead that does not break `read-after-write` consistency.

FYI: @findepi

### PR DESCRIPTION
**Pull request by Praveen2112**
Fixes #103. We remove exists check in PrestoS3FileSystem when we try to create a file in non-overwrite mode as adding the exists check breaks the S3 read-after-write consistency.

### CODE REVIEW

**Comment by Praveen2112** [comment]
@sopel39 , @electrum

File: presto-hive/src/main/java/io/prestosql/plugin/hive/s3/PrestoS3FileSystem.java [hunk 1]
```diff
@@ -360,10 +360,7 @@ public FSDataInputStream open(Path path, int bufferSize)
     public FSDataOutputStream create(Path path, FsPermission permission, boolean overwrite, int bufferSize, short replication, long blockSize, Progressable progress)
             throws IOException
     {
-        if ((!overwrite) && exists(path)) {
-            throw new IOException("File already exists:" + path);
-        }
-
+        //HACK: If overwrite is false then we should check if the file already exists but doing so might break S3 read-after-write consistency so skipping it.
```
Comment (sopel39) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/s3/PrestoS3FileSystem.java]: I would add `TODO` here. Even previously this code was only best effort since file could have been created by other process after existence check.

nit: add space after `//`

Comment (Praveen2112) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/s3/PrestoS3FileSystem.java]: Updated it

File: presto-hive/src/main/java/io/prestosql/plugin/hive/s3/PrestoS3FileSystem.java [hunk 2]
```diff
@@ -360,10 +360,8 @@ public FSDataInputStream open(Path path, int bufferSize)
     public FSDataOutputStream create(Path path, FsPermission permission, boolean overwrite, int bufferSize, short replication, long blockSize, Progressable progress)
             throws IOException
     {
-        if ((!overwrite) && exists(path)) {
-            throw new IOException("File already exists:" + path);
-        }
-
+        // HACK: If overwrite is false then we should check if the file already exists but doing so might break S3 read-after-write consistency so skipping it.
```
Comment (sopel39) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/s3/PrestoS3FileSystem.java]: nit: remove empty line above `HACK` and add empty line after `TODO` (so it doesn't look related to `if (!stagingDirectory.exists()) {`

Comment (electrum) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/s3/PrestoS3FileSystem.java]: How about changing this to
```java
// Ignore the overwrite flag, since Presto always writes to unique file names.
// Checking for file existence can break read-after-write consistency.
```
I don't like adding a TODO here since we are unlikely to implement this unnecessary-for-Presto behavior later.

Comment (Praveen2112) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/s3/PrestoS3FileSystem.java]: I have updated it.
